libpod/events/logfile.go

```diff
@@ -107,3 +115,125 @@ func (e EventLogFile) Read(ctx context.Context, options ReadOptions) error {
 func (e EventLogFile) String() string {
 	return LogFile.String()
 }
+
+// Rotates the log file if the log file size and content exceeds limit
+func rotateLog(logfile string, content string, limit uint64) (bool, error) {
+	e := &Event{}
+	if limit == 0 {
+		return false, nil
+	}
+	file, err := os.Stat(logfile)
+	if err != nil {
+		return false, err
+	}
+	var filesize = uint64(file.Size())
+	var contentsize = uint64(len([]rune(content)))
+	if filesize+contentsize >= limit {
+		if err := truncate(logfile); err != nil {
+			return false, err
+		}
+		rotateEvent := e.RotateEvent(Rotate)
+		rotateEventJSONString, err := json.Marshal(rotateEvent)
+		if err != nil {
+			return false, nil
+		}
+		rotateEventString := string(rotateEventJSONString)
+		f, err := os.OpenFile(logfile, os.O_APPEND|os.O_CREATE|os.O_WRONLY, 0700)
+		if err != nil {
+			return false, err
+		}
+		if _, err := f.WriteString(fmt.Sprintf("%s\n", rotateEventString)); err != nil {
+			return false, err
+		}
+		return true, nil
+	}
+	return false, nil
+}
+
+// Truncates the log file and saves 50% of content to new log file
+func truncate(path string) error {
+	orig, err := os.Open(path)
+	if err != nil {
+		return err
+	}
+
+	origFinfo, err := orig.Stat()
+	if err != nil {
+		return err
+	}
+
+	size := origFinfo.Size()
+	threshold := uint64(size / 2)
+
+	tmp, err := ioutil.TempFile("", "")
+	if err != nil {
+		return err
+	}
+	var readBytes uint64
+	scanner := bufio.NewScanner(orig)
+	for scanner.Scan() {
+		rawLine := scanner.Bytes()
+		rawLine = append(rawLine, '\n')
+		readBytes += uint64(len(rawLine))
+		if readBytes <= threshold {
+			// Skip the line if we're still below 50% of the file.
+			continue
+		}
+		if _, err := tmp.Write(rawLine); err != nil {
+			return fmt.Errorf("writing to temporary file: %w", err)
+		}
+	}
+	if err := orig.Close(); err != nil {
+		return err
+	}
+	if err := tmp.Close(); err != nil {
+		return err
+	}
+	return renameLog(tmp.Name(), path)
+}
+
+// Renames from, to
+func renameLog(from, to string) error {
+	err := os.Rename(from, to)
+	if err == nil {
+		return nil
+	}
+
+	if !strings.Contains(err.Error(), "invalid cross-device link") {
+		return err
+	}
```

Comment:
errors should not be compared with string unless absolutely necessary. 
use `errors.Is()` I think you want to match against `unix.EXDEV`